### PR TITLE
Add support for servicing (post-build.yml)

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Blocked by https://github.com/dotnet/arcade/pull/3572

cc @JohnTortugo @mmitche 

@dotnet/coreclr-infra

### CODE REVIEW

**Comment by danmosemsft** [comment]
we're trying to use a single area label or the labeler bot will get confused

File: eng/test-job.yml [hunk 1]
```diff
@@ -330,12 +330,12 @@ jobs:
...
+          artifactName: ${{ format('TestLogs{0}_{1}_{2}_{3}_{4}', '_corefx',     parameters.osIdentifier, parameters.archType, parameters.buildConfig, parameters.testGroup) }}
         ${{ if and(ne(parameters.corefxTests, true), eq(parameters.readyToRun, true)) }}:
-          artifactName: ${{ format('test{0}_{1}_{2}_{3}_{4}_Logs', '_r2r',        parameters.osIdentifier, parameters.archType, parameters.buildConfig, parameters.testGroup) }}
+          artifactName: ${{ format('TestLogs{0}_{1}_{2}_{3}_{4}', '_r2r',        parameters.osIdentifier, parameters.archType, parameters.buildConfig, parameters.testGroup) }}
         ${{ if and(ne(parameters.corefxTests, true), ne(parameters.readyToRun, true)) }}:
-          artifactName: ${{ format('test{0}_{1}_{2}_{3}_{4}_Logs', '',            parameters.osIdentifier, parameters.archType, parameters.buildConfig, parameters.testGroup) }}
+          artifactName: ${{ format('TestLogs_{0}_{1}_{2}_{3}',                   parameters.osIdentifier, parameters.archType, parameters.buildConfig, parameters.testGroup) }}
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply eng/test-job.yml]: That makes the strings unfortunately asymmetric, which is why it was done the way it was. FWIW.

**Comment by ViktorHofer** [comment]
@AaronRobinsonMSFT will this also fix the publish to BAR failures in official builds? https://dnceng.visualstudio.com/internal/_build/results?buildId=300175

**Comment by AaronRobinsonMSFT** [comment]
@ViktorHofer I don't really know the answer to that. If I had to guess, I would say 'yes' because this is  the official Arcade approach. Perhaps @elinor-fung or @mmitche could speak to that problem and how it relates to this work.

**Comment by mmitche** [comment]
@ViktorHofer I don't know what that exact failure is, but it's quite possible this could get rid of it (by working around it entirely)

**Comment by ViktorHofer** [comment]
@AaronRobinsonMSFT any idea what's causing some legs (ie PublishAssets in .NET Tools - Master) to have an "abandoned" state? Unfortunately we still don't see a new coreclr build in corefx.

cc @stephentoub

**Comment by AaronRobinsonMSFT** [comment]
@ViktorHofer Can you point me to the failing issue? Is this in corefx or coreclr. Note that CoreCLR hasn't been succeeding in quite some time and the current official build issues in CoreCLR seem to be related to missing NuGet feeds so like my comment in the referenced issue, I don't see how the connection is being made to this work.

**Comment by ViktorHofer** [comment]
> Can you point me to the failing issue?

https://dnceng.visualstudio.com/internal/_build/results?buildId=305428

> I don't see how the connection is being made to this work.

I did not intend to criticize your work here at all. I just thought as you introduced the new templates you might have context on the "abandoned" states.

**Comment by AaronRobinsonMSFT** [comment]
Ah, I see the connection now. Those steps are defined by Arcade and execute when the build succeeds - at least I think that is the trigger. In this case tests are failing for the above mentioned issue with NuGet so I don't know if they are expected to run.

Note that any stages/jobs outside of the "Build" stage are defined by Arcade and I would expect any repo consuming them to hit similar issues. Do you know if CoreFX has moved over to Arcade publishing?

@JohnTortugo @mmitche Can either of you comment on why the publishing steps in the reference build would be marked "abandoned"? Is this related to the failing tests in that leg? I would have instead expected them to be marked as "skipped"

**Comment by ViktorHofer** [comment]
> Do you know if CoreFX has moved over to Arcade publishing?

Yes we have and that's one of our builds where the legs are run and marked correctly: https://dnceng.visualstudio.com/internal/_build/results?buildId=305712.

**Comment by AaronRobinsonMSFT** [comment]
Thanks for the link. This is how it was running for me during testing when no tests were running. This means we need to verify that tests don't cause subsequent jobs to get abandoned. I will start a thread offline and include you so we can get to the bottom of the issue.

**Pull request merged**